include/services/OswServiceTaskBLEServer.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+#ifdef OSW_FEATURE_BLE_SERVER
+#include "osw_service.h"
+#include "services/NotifierClient.h"
+
+#define CONFIG_BT_NIMBLE_ROLE_PERIPHERAL
+#include <NimBLEDevice.h>
+
+class OswServiceTaskBLEServer : public OswServiceTask {
+  public:
+    OswServiceTaskBLEServer() {};
+    virtual void setup() override;
+    virtual void loop() override;
+    virtual void stop() override;
+    ~OswServiceTaskBLEServer() {};
+
+    void enable();
+    void disable();
+    bool isEnabled();
+
+    void updateName();
+  private:
+    class ServerCallbacks: public NimBLEServerCallbacks {
+      public:
+        ServerCallbacks(OswServiceTaskBLEServer* task): task(task) {};
+
+      private:
+        void onConnect(BLEServer* pServer);
+        void onDisconnect(BLEServer* pServer);
+        uint32_t onPassKeyRequest();
+        bool onConfirmPIN(uint32_t pass_key) {
+            return false; // we only report display-only
+        };
+
+        OswServiceTaskBLEServer* task;
+    };
+    class BatteryLevelCharacteristicCallbacks: public NimBLECharacteristicCallbacks {
+        void onRead(NimBLECharacteristic* pCharacteristic);
+        uint8_t byte = 0; // will be read from
```

Comment:
maybe `byte` -> `batteryLevel`.

Reply:
In contrary: We had this discussion before, if an explicit name does yield additional valuable information, it should be added. But in these classes, these classes all have one member and are updating it upon read and then provided it to the transmitter. Their names are very clearly stating what they do, hence I think we can save the additional typing on the dev and just call the data as it is being interpreted. Also, explicit names prevent easy copy-pasting :wink:

Reply:
Good point of view.